src/specgen.ts

```diff
@@ -1 +1,138 @@
+}
+
+ast.addExistingSourceFiles( providedSourcePaths );
+ast.getSourceFiles().forEach(sourceFile => {
+    const sourceClasses = sourceFile.getClasses();
+
+    if (!sourceClasses.length) {
+        return;
+    }
+
+    const sourcePath = sourceFile.getFilePath();
+    const outputPath = sourcePath.replace( srcDir, outDir ).replace(/\.ts$/, specFilenameSuffix);
+    const outputDirname = path.dirname(outputPath);
+    const relativePath = path.relative(outputDirname, sourcePath).replace( /\\/g, '/' );
+
+    const specImports = sourceClasses
+        .map(sourceClass => {
+            return `import { ${sourceClass.getName()} } from '${relativePath.replace(/\.ts$/, '')}';`
+        });
+
+    const generatedSpec = sourceClasses
+        .map(sourceClass => {
+            return [
+                ...createSpecHeaderForClass(sourceClass),
+                ...prefix( tab, createSpecOutline( sourceClass )),
+                ...createSpecFooterForClass(sourceClass),
+            ];
+        })
+        .reduce(( acc: string[], next: string[] ) => {
+            return acc.concat(next);
+        }, []);
+
+    const outputSpec = [
+        ...specImports,
+        '',
+        ...generatedSpec,
+        '',
+    ].join('\n');
+
+    try {
+        statSync( outputDirname );
+    } catch ( err ) {
+        mkdirp.sync( outputDirname );
+    }
+
+    writeFileSync( outputPath, outputSpec );
+    console.info( outputPath );
+});
+
+function prefix(str: string, lines: string[]): string[] {
+    return lines.map(line => line !== '' ? str + line : '');
+}
+
+function createSpecHeaderForClass(sourceClass: ClassDeclaration): string[] {
+    const className = sourceClass.getName();
+    return [`describe( '${className}', () => {`];
+}
+
+function createSpecFooterForClass(sourceClass: ClassDeclaration): string[] {
+    return ['',`});`];
+}
+
+function createDescribeBlock(classMember: MethodDeclaration | GetAccessorDeclaration | SetAccessorDeclaration, body: string[] = ['']): string[] {
+    const lines = [''];
+    const memberName = classMember.getName();
+    lines.push(
+        `describe( '${memberName}', () => {`,
+        ...prefix( tab, body ),
+        `});`
+    );
+    return lines;
+}
+
+function createSpecOutline(sourceClass: ClassDeclaration): string[] {
+    const lines: string[] = [];
+    const classMembers = sourceClass.getMembers();
+    classMembers.forEach( classMember => {
+
+        if ( classMember instanceof MethodDeclaration 
+            || classMember instanceof GetAccessorDeclaration
+            || classMember instanceof SetAccessorDeclaration )
+        {
+            if ( classMember.getScope() === Scope.Protected
+                || classMember.getScope() === Scope.Public 
+            ) {
+                lines.push( ...createDescribeBlock( classMember ));
+            }        
+        } 
+    });
+    return lines;
+}
+
+
+function getRespectiveSpecFile(sourceClass: ClassDeclaration): SourceFile | undefined {
```

Comment:
This function is not used. For now, if the spec file already exists, it should not replace it.